File: 485/Solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ccnt = 0
        maxcnt = 0
        for i in nums:
            ccnt = ccnt * i + i
            maxcnt = maxcnt if ccnt < maxcnt else ccnt

        return maxcnt

File: 485/Solution.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int cons = 0;
        int maxCount = 0;
        for(int x: nums) {
            cons = cons*x+x;
            if (cons > maxCount) {
                maxCount = cons;
            }
        }
        return maxCount;
    }
}